ProbeSet sample data editing

zsloan opened the pull request:
This is meant to add ProbeSet sample data editing. Still in progress for now.

I'm not sure why it includes a bunch of other random unrelated commits from the past month.

fredmanglis committed:
Provide the client_id as part of the post request.

BonfaceKilz committed:
Trigger CI

fredmanglis committed:
Revert "Skip tests broken by Python-3.10 temporarily"

This reverts commit f52247c15f3694f3dd5fd0fd79c3e15376137e07.

fredmanglis committed:
Move documentation comment to beginning of function

fredmanglis committed:
Bug: Retrieve configs from app, not environment

Retrieve the configuration values from the application object, not from the
environment. We are assured of having the configuration values set in the
application - we do not have that assurance for the environment.

fredmanglis committed:
Bug: Set the `data_exists` variable within the context manager

The `cursor.fetchone()` call that was used as the condition to the `if` was
called outside of the context manager, and therefore would always give a
non-truthy value at best and an inconsistent result at worst.

This commit gets the value before the context manager has exited and stores it
for later use.

fredmanglis committed:
Fix dict key name

Remove extra colon (:) at the end of the name that was leading to the number
of cases not being presented as expected.

fredmanglis committed:
Bug: Fetch group name by dataset_id

Change the code to fetch the group name by the dataset ID, since according to
usage[1] of the `retrieve_group_name` function, the value passed in is the
`dataset_id` and not the `group_id`.

Change the name from `retrieve_group_name` to `retrieve_phenotype_group_name`
to more clearly indicate that this function concerns itself with the groups
that relate to phenotypes.

[1](https://github.com/genenetwork/genenetwork2/blob/1bbb0430732b7fa5102d7dcbda80ebda252f5424/wqflask/wqflask/metadata_edits.py)

fredmanglis committed:
Fetch single phenotype trait by `dataset_id` and `trait_name`.

fredmanglis committed:
Fetch metadata for a single phenotype.

fredmanglis committed:
Fetch a single publication by `PublicationId`

fredmanglis committed:
Fix wrong import, and typing issues.

fredmanglis committed:
Bug: Change from `_id` to `id_` to avoid downstream breakage.

The downstream code expects `id_` not `_id`.

fredmanglis committed:
Extract reusable code to separate module

Extract the reusable function into a separate `query_tools` module for use in
other modules.

fredmanglis committed:
Extract code dealing with ProbeSets to their own module.

arunisaac committed:
Make repository a guix channel.

* .guix-channel: New file.
* guix.scm: Move to ...
* .guix/genenetwork3-package.scm: ... here. Convert into a module with public
package variables.

fredmanglis committed:
Rename function: fetch_publication ==> fetch_publication_by_id

fredmanglis committed:
Fetch a publication by its PubMed ID

fredmanglis committed:
Update a Publication in the database.

fredmanglis committed:
Add some debug utilities.

fredmanglis committed:
Function to create a metadata audit trail.

BonfaceKilz committed:
Roll back to python 3.9 in .guix-channel

BonfaceKilz committed:
Merge pull request #126 from BonfaceKilz/fix/pin-python-3.9

Roll back to python 3.9 in .guix-channel

fredmanglis committed:
Fetch `metadata_audit` trail with direct query functions.

fredmanglis committed:
Remove debug statement.

fredmanglis committed:
Fix variable name.

fredmanglis committed:
Paginate the resource data items.

fredmanglis committed:
Fix bugs. Return resource without data items.

* Fix bugs introduced while attempting pagination.
* Return resource object without attached data items.

fredmanglis committed:
Fix linting and typing issues.

fredmanglis committed:
Fix typing issue

fredmanglis committed:
Assign `group-leader` role to the admin for automatic group.

fredmanglis committed:
Fix module name.

fredmanglis committed:
Use correct variable

After copying the calls to assign appropriate roles to the admin in the
script, I forgot to update the variable name. This commit fixes that.

fredmanglis committed:
Integrate Temp traits to Auth

Temporary traits are not saved to the database, and have no attached
resource (at this time). As such, make them all simply public-read to enable
access for users.

This might change after more information on usage of temporary traits is
collected and considered.

fredmanglis committed:
Remove ORM-dependent `fetchall` and `fetchone` functions

Remove Object-Relational Mapping dependent function, `fetchall` and `fetchone`
so as to prevent theirs use in the code moving forward.

fredmanglis committed:
Remove ORM-dependent `insert` function

Remove the object-relation-mapping dependent `insert` function to prevent it
being used in the code down the line.

fredmanglis committed:
Add deprecation notice to ORM-dependent `update` function

Add a deprecation notice to discourage other devs from using the deprecated
`update` function.

zsloan committed:
Make several sample_data fetching functions specific to phenotype or mrna assay resources, and change their naming accordingly

zsloan committed:
Add function for retrieving mRNA Assay group name

zsloan committed:
Remove Id select from probeset metadata query, since it's included in cols

zsloan committed:
Fix mrna group name query

zsloan committed:
Fix get_mrna_sample_data query

fredmanglis committed:
Remove the ORM-dependent `update` function.

fredmanglis committed:
Remove tests for deleted function.

fredmanglis committed:
Remove unused import

zsloan committed:
Fix query for retrieving mrna dataset group name

BonfaceKilz commented:
zsloan ***@***.***> aliandika:

> This is meant to add ProbeSet sample data editing. Still in progress for now.
>
> I&#39;m not sure why it includes a bunch of other random unrelated commits from the past month.
> You can view, comment on, or merge this pull request online at:

Try rebasing.  Do any of those commits interfere?
If not, don't worry too much about it.

>   https://github.com/genenetwork/genenetwork3/pull/127
>
> -- Commit Summary --
>
>   * Provide the client_id as part of the post request.
>   * Trigger CI
>   * Revert &quot;Skip tests broken by Python-3.10 temporarily&quot;
>   * Move documentation comment to beginning of function
>   * Bug: Retrieve configs from app, not environment
>   * Bug: Set the `data_exists` variable within the context manager
>   * Fix dict key name
>   * Bug: Fetch group name by dataset_id
>   * Fetch single phenotype trait by `dataset_id` and `trait_name`.
>   * Fetch metadata for a single phenotype.
>   * Fetch a single publication by `PublicationId`
>   * Fix wrong import, and typing issues.
>   * Bug: Change from `_id` to `id_` to avoid downstream breakage.
>   * Extract reusable code to separate module
>   * Extract code dealing with ProbeSets to their own module.
>   * Make repository a guix channel.
>   * Rename function: fetch_publication ==&gt; fetch_publication_by_id
>   * Fetch a publication by its PubMed ID
>   * Update a Publication in the database.
>   * Add some debug utilities.
>   * Roll back to python 3.9 in .guix-channel
>   * Merge pull request #126 from BonfaceKilz/fix/pin-python-3.9
>   * Function to create a metadata audit trail.
>   * Fetch `metadata_audit` trail with direct query functions.
>   * Remove debug statement.
>   * Fix variable name.
>   * Paginate the resource data items.
>   * Fix bugs. Return resource without data items.
>   * Fix linting and typing issues.
>   * Fix typing issue
>   * Assign `group-leader` role to the admin for automatic group.
>   * Fix module name.
>   * Use correct variable
>   * Integrate Temp traits to Auth
>   * Remove ORM-dependent `fetchall` and `fetchone` functions
>   * Remove ORM-dependent `insert` function
>   * Add deprecation notice to ORM-dependent `update` function
>   * Make several sample_data fetching functions specific to phenotype or mrna assay resources, and change their naming accordingly
>   * Add function for retrieving mRNA Assay group name
>   * Remove Id select from probeset metadata query, since it&#39;s included in cols
>   * Fix mrna group name query
>   * Fix get_mrna_sample_data query
>
> -- File Changes --
>
>     A .guix-channel (23)
>     A .guix/genenetwork3-package.scm (19)
>     M gn3/auth/authentication/oauth2/views.py (4)
>     M gn3/auth/authorisation/data/views.py (18)
>     M gn3/auth/authorisation/resources/models.py (64)
>     M gn3/auth/authorisation/resources/views.py (35)
>     M gn3/db/__init__.py (75)
>     M gn3/db/datasets.py (43)
>     M gn3/db/metadata_audit.py (49)
>     M gn3/db/phenotypes.py (121)
>     A gn3/db/probesets.py (77)
>     A gn3/db/query_tools.py (7)
>     M gn3/db/sample_data.py (107)
>     M gn3/db_utils.py (2)
>     A gn3/debug.py (10)
>     M gn3/templates/oauth2/authorise-user.html (3)
>     D guix.scm (39)
>     A guix.scm (1)
>     M pytest.ini (2)
>     M scripts/migrate_existing_data.py (4)
>     D tests/unit/db/test_audit.py (31)
>     M tests/unit/db/test_db.py (73)
>     M tests/unit/db/test_species.py (3)
>
> -- Patch Links --
>
> https://github.com/genenetwork/genenetwork3/pull/127.patch
> https://github.com/genenetwork/genenetwork3/pull/127.diff
>
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/genenetwork/genenetwork3/pull/127
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***&gt;

-- 
(Life is like a pencil that will surely run out,
    but will leave the beautiful writing of life.)
(D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
    (hkp://keys.openpgp.org))

BonfaceKilz commented:
zsloan ***@***.***> aliandika:

> This is meant to add ProbeSet sample data editing. Still in progress for now.
>
> I&#39;m not sure why it includes a bunch of other random unrelated commits from the past month.
> You can view, comment on, or merge this pull request online at:
>

Just had a quick skim on the unrelated commits:
┌────
│    * Trigger CI
│    * Revert &quot;Skip tests broken by Python-3.10 temporarily&quot;
│    * Roll back to python 3.9 in .guix-channel
│    * Merge pull request #126 from BonfaceKilz/fix/pin-python-3.9
└────

And those were put in place to fix gn2 to
python3.9 in our CI/CD set-up.

-- 
(Life is like a pencil that will surely run out,
    but will leave the beautiful writing of life.)
(D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
    (hkp://keys.openpgp.org))

zsloan committed:
Fix probeset sample data and CSV sample data functions

zsloan committed:
Implement ProbeSet sample data editing

Currently there's a lot of duplicated code, so I need to ask how best to simplify it later

zsloan committed:
Fix a few misnamed tables in mrna delete/insert functions

fredmanglis committed:
auth: Remove documentation

The authentication/authorisation stuff is being moved to a separate
repository. This commit removes the documentation for how to run migrations
from this repository since it has been moved over to the new repository.

zsloan committed:
Fix fetch_trait query for phenotypes

zsloan committed:
Fix queries for fetching group names

zsloan committed:
Combine pheno/mrna update functions + some other fixes

zsloan committed:
Fix tests to work with changes

zsloan committed:
Fix a couple remaining issues with function parameters

zsloan committed:
Fix query for fetching DataId when inserting ProbeSet data
